src/hazelcore/Action_Exp.re

```diff
@@ -1087,6 +1087,34 @@ and syn_perform_opseq =
 
   /* Deletion */
 
+  /* This case prevents deletion of parentheses from the front with backspace
+     or from the rear with delete from taking two keypresses, the first of
+     which does not result in caret movement. */
+  | (
+      a,
+      ZOperand(
+        ParenthesizedZ((
+          [],
+          ExpLineZ(ZOpSeq(_, ZOperand(zop, (prefix, suffix)))),
+          [],
+        )),
+        (upper_prefix, upper_suffix),
+      ),
+    )
+      when
+        ZExp.is_before_zoperand(zop)
```

Comment:
this isn't correct -- zop might not be the leftmost position if prefix isn't empty. you should check is_before on the entire opseq. manifests as wrong behavior when you do, e.g. 1 + (2 + |3 + 4)

Reply:
Ah whoops. Though couldn't I just change the prefix pattern to E and keep the check the same?

Reply:
or I guess that would require separate logic for delete... i'll do it your way

Reply:
fixed